runtime/parachains/src/session_info/migration.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! A module that is responsible for migration of storage.
+
+use crate::session_info::{Config, Pallet, Store};
+use frame_support::{pallet_prelude::*, traits::StorageVersion, weights::Weight};
+
+/// The current storage version.
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(2);
+
+/// Migrates the pallet storage to the most recent version, checking and setting the `StorageVersion`.
+pub fn migrate_to_latest<T: Config>() -> Weight {
+	let mut weight = 0;
+	if StorageVersion::get::<Pallet<T>>() <= 1 {
+		weight += migrate_to_v2::<T>();
+		StorageVersion::new(2).put::<Pallet<T>>();
+	}
+	weight
+}
+
+pub fn migrate_to_v2<T: Config>() -> Weight {
+	let mut vs = 0;
+
+	<Pallet<T> as Store>::Sessions::translate_values(|old: primitives::v1::SessionInfo| {
+		vs += 1;
+		Some(primitives::v2::SessionInfo::from(old))
```

Comment:
Nit: Don't you think it would be better to avoid using `from` and describe translation here? I think it would make it easier to reason about the actual work done by the migration

Reply:
Not sure. The translation is also used for when the runtime api for version 2 fails and it uses version 1 with the conversion to 2. Are you suggesting to duplicate the code here for clarity?

Reply:
Nope, sorry. I was almost sure it was only used here.